lib/iex/test/iex/helpers_test.exs

```diff
@@ -49,6 +49,22 @@ defmodule IEx.HelpersTest do
     assert capture_io(fn -> h __info__ end) == "No documentation for __info__ was found\n"
   end
 
+  test "h helper for callbacks" do
+    with_file ["a_behaviour.ex", "impl.ex"], [behaviour_module, impl_module], fn ->
```

Comment:
There seems to be maybe a small bug with using protocols like you mentioned. I tried it with `String.Chars` by adding `@doc @moduledoc` [here](https://github.com/mgwidmann/elixir/blob/docs_for_callbacks/lib/elixir/lib/string/chars.ex#L16) and it works fine when I try `h String.Chars.to_string`. But when doing `h String.Chars.Atom.to_string` I get a blank doc. I suspect this is because the `String.Chars.to_string` is stored using `def` which puts it into `:docs` instead of into `:callback_docs` where we are expecting to find it when we fallback.

Should I be looking up in `:docs` when `:callback_docs` has nothing?  [here is where I mean we do the wrong lookup](https://github.com/mgwidmann/elixir/blob/docs_for_callbacks/lib/elixir/lib/code.ex#L643)